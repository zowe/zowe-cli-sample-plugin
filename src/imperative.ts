/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {IImperativeConfig} from "@brightside/imperative";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    overrides: {
        CredentialManager: "./overrides/CredentialManager.override"
    },
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI sample plug-in",
    pluginAliases: ["zcsp"],
    rootCommandDescription: "Welcome to the sample plug-in for Zowe CLI!\n\n The sample plug-in " +
        "(& CLI) follows the Zowe CLI command syntax 'zowe [group] [action] [object] [options]'. " +
        "Where, in the case of the plugin, " +
        "the [group] is the package.json name, " +
        "the [actions] are defined in the project under 'src/cli/', " +
        "& the [objects] (definitions & handler) are defined in the project under 'src/cli/'.",
    productDisplayName: "Zowe CLI Sample Plug-in",
    name: "zowe-cli-sample"
};

export = config;
