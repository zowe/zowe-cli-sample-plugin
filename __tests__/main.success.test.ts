/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { Imperative } from "@brightside/imperative";
// The main tests exist in seperate files because of module resolution in Jest,
// we want to require the main module a few times to perform different tests
describe("main success", () => {
    it("should log initialization messages", (done) => {
        let logMsgs: string = "";
        let callCount: number = 0;

        // "Mock" the api object - which is a readonly static within Imperative
        Object.defineProperty(Imperative, "api", {
            value: {
                imperativeLogger: {
                    info: jest.fn((msg) => {
                        logMsgs += " " + msg;
                        callCount++;
                        if (callCount === 2) {
                            expect(logMsgs).toMatchSnapshot();
                            done();
                        }
                    })
                }
            }
        });

        // Mock imperative init
        Imperative.init = jest.fn((args): Promise<void> => {
            return new Promise<void>((fulfill, reject) => {
                fulfill();
            });
        });

        // Mock imperative parse
        Imperative.parse = jest.fn(() => {
            // Do nothing
        });

        // Require should run the init/parse
        const main = require("../src/main");
    });
});
