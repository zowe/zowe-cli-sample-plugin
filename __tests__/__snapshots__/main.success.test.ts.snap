// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main success should log initialization messages 1`] = `" Imperative Initialized the sample plugin CLI! Parsed the input command!"`;
